### PR TITLE
refine NER.

### PR DESCRIPTION
**Pull request by lcy-seso**
refactor NER demo.

### CODE REVIEW

**Comment by lcy-seso** [comment]
README hasn't updated yet.

File: sequence_tagging_for_ner/network_conf.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    word_caps_vector = paddle.layer.concat(input=emb_layers)
+
+    mix_hidden_lr = 1e-3
+    rnn_para_attr = paddle.attr.Param(initial_std=0.0, learning_rate=0.1)
+    hidden_para_attr = paddle.attr.Param(
+        initial_std=1 / math.sqrt(hidden_dim), learning_rate=mix_hidden_lr)
+
+    forward_hidden, rnn_forward = stacked_rnn(word_caps_vector, hidden_dim,
+                                              hidden_para_attr, rnn_para_attr)
+    backward_hidden, rnn_backward = stacked_rnn(
```
Comment (guoshengCS) [inline sequence_tagging_for_ner/network_conf.py]: It seems that the two `stacked_rnn` branches import separated hidden layers at the first level, while the two RNN branches in the raw [config](https://github.com/lcy-seso/paddle_confs_v1/blob/master/sequence_tagging_for_ner/ner.conf) share the same hidden layer. It matters if strict consistency is required.

File: sequence_tagging_for_ner/infer.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def _infer_a_batch(inferer, test_data, id_2_word, id_2_label):
+        probs = inferer.infer(input=test_data, field=["id"])
+        assert len(probs) == sum(len(x[0]) for x in test_data)
+
+        for idx, test_sample in enumerate(test_data):
+            start_id = 0
+            pred_str = ""
+            for w, tag in zip(test_sample[0],
+                              probs[start_id:start_id + len(test_sample[0])]):
+                pred_str += "%s[%s] " % (id_2_word[w], id_2_label[tag])
```
Comment (guoshengCS) [inline sequence_tagging_for_ner/infer.py]: Since lowercase is used, the outputs might be different from the raw texts. It should be noticed if the raw texts are wanted.

File: sequence_tagging_for_ner/network_conf.py (hunk 1 above)
Comment (lcy-seso) [reply sequence_tagging_for_ner/network_conf.py]: This is a bug, these two RNN should not share the parameters. I will fix this. Thanks for your comment.

Comment (lcy-seso) [reply sequence_tagging_for_ner/network_conf.py]: I modify the network configuration to keep it consistent with the original one.

File: sequence_tagging_for_ner/README.md [hunk 3]
```diff
@@ -28,51 +45,47 @@ NER任务的输入是"一句话"，目标是识别句子中的实体边界及类
...
+Ekeus        NNP  I-NP  I-PER
+heads        VBZ  I-VP  O
+for          IN   I-PP  O
+Baghdad      NNP  I-NP  I-LOC
+.            .    O     O
 ```
 
-其中第一列为原始句子序列（第二、三列分别为词性标签和句法分析中的语块标签，这里暂时不用），第四列为采用了I-TYPE方式表示的NER标签（I-TYPE和BIO方式的主要区别在于语块开始标记的使用上，I-TYPE只有在出现相邻的同类别实体时对后者使用B标记，其他均使用I标记），句子之间以空行分隔。
+- 第一列为原始句子序列
+- 第二、三列分别为词性标签和句法分析中的语块标签，本利不使用
```
Comment (guoshengCS) [inline sequence_tagging_for_ner/README.md]: "本利不使用"应改为"本例不使用"

File: sequence_tagging_for_ner/README.md [hunk 4]
```diff
@@ -28,51 +45,47 @@ NER任务的输入是"一句话"，目标是识别句子中的实体边界及类
...
-|---|---|
-| eng.train | 训练数据 |
-| eng.testa | 验证数据，可用来进行参数调优 |
-| eng.testb | 评估数据，用来进行最终效果评估 |
++ 我们仅在`data`目录下的`train`和`test`文件中放置少数样本用以示例输入数据格式。
++ 本例依赖数据还包括
+    1. 输入文本的词典
+    2. 为词典中的词语提供预训练好的词向量
+    2. 标记标签的词典
+   标记标签词典已附在`data`目录中，对应于`data/target.txt`文件。输入文本的词典以及词典中词语的预训练的词向量来自：[Stanford CS224d](http://cs224d.stanford.edu/)课程作业。**为运行本例，请首先在`data`目录下运行`download.sh`脚本下载预训练的词向量。** 完成后会将这两个文件一并放入`data`目录下，输入文本的词典和预训练的词向量分别对应：`data/vocab.txt`和`data/wordVectors.txt`这两个文件。
```
Comment (guoshengCS) [inline sequence_tagging_for_ner/README.md]: "下载预训练的词向量"是否应为"下载输入文本的词典和预训练的词向量"

**Review by guoshengCS — APPROVED** [review]

File: sequence_tagging_for_ner/README.md (hunk 4 above)
Comment (lcy-seso) [reply sequence_tagging_for_ner/README.md]: done

File: sequence_tagging_for_ner/README.md (hunk 3 above)
Comment (lcy-seso) [reply sequence_tagging_for_ner/README.md]: done

**Review by lcy-seso — COMMENTED** [review]
follow comments, thank you.

File: sequence_tagging_for_ner/infer.py (hunk 2 above)
Comment (lcy-seso) [reply sequence_tagging_for_ner/infer.py]: This hasn't be fixed yet. During inferring only normalized text is printed.
